django_components/templatetags/component_tags.py

```diff
@@ -160,99 +162,177 @@ def __init__(self, slot: "SlotNode", context: Context):
 
+    # (lemontheme): If 'default' is accepted as option keyword for slot tags,
+    #  then I propose 'original' replaces 'default' in the 'slot.super' sense.
+    #  Otherwise, we would have both
+    #  - {% slot ... default %} ...
+    #  and
+    #  - {% fill ... as <alias> %} <alias>.default {%endfill %}.
+    #  I'm afraid this risks being a source of confusion, as users might wrongly assume
+    #  that there is a link between the two separate usages of 'default'.
+    @property
+    def original(self) -> str:
+        return self.default
+
 
 class SlotNode(Node):
     def __init__(
-        self, name, nodelist, template_name: str = "", required=False
+        self,
+        name: str,
+        nodelist: NodeList,
+        template_name: str = "",
+        is_required: bool = False,
+        is_default: bool = False,
     ):
         self.name = name
         self.nodelist = nodelist
         self.template_name = template_name
-        self.is_required = required
+        self.is_required = is_required
+        self.is_default = is_default
+
+    @property
+    def active_flags(self):
+        m = []
+        if self.is_required:
+            m.append("required")
+        if self.is_default:
+            m.append("default")
+        return m
 
     def __repr__(self):
-        return f"<Slot Node: {self.name}. Contents: {repr(self.nodelist)}>"
+        return f"<Slot Node: {self.name}. Contents: {repr(self.nodelist)}. Flags: {self.active_flags}>"
 
     def render(self, context):
         if FILLED_SLOTS_CONTEXT_KEY not in context:
             raise TemplateSyntaxError(
                 f"Attempted to render SlotNode '{self.name}' outside a parent component."
             )
-        filled_slots: Dict[str, List[FillNode]] = context[
+        # Try to find FillNode
+        filled_slots: Dict[Optional[str], List[FillNode]] = context[
             FILLED_SLOTS_CONTEXT_KEY
         ]
         fill_node_stack = filled_slots.get(self.name, None)
-        extra_context = {}
-        if not fill_node_stack:  # if None or []
-            nodelist = self.nodelist
-            # Raise if slot is 'required'
-            if self.is_required:
+        fill_node: Optional[FillNode] = None
+        if fill_node_stack:  # if None or []
+            fill_node = fill_node_stack.pop()
+        elif self.is_default:
+            implicit_fill_key = None
+            implicit_fill_node_stack = filled_slots.get(implicit_fill_key)
+            if implicit_fill_node_stack:
+                fill_node = implicit_fill_node_stack.pop()
+        if self.is_required:
+            if fill_node is None:
                 raise TemplateSyntaxError(
                     f"Slot '{self.name}' is marked as 'required' (i.e. non-optional), "
                     f"yet no fill is provided. Check template '{self.template_name}'"
                 )
-        else:
-            fill_node = fill_node_stack.pop()
+        # //
```

Comment:
Strange comment, remove?

Reply:
Will do. (It's my way of saying 'here ends the block of code up until which the previous comment holds'.)